### PR TITLE
Allow comments between newlines in chaining hints

### PR DESCRIPTION
**Pull request by unratito**
Currently, chaining hints are not generated if there are comments between newlines, which is a very common pattern:

```rust
let vec = (0..10)
    // Multiply by 2
    .map(|x| x * 2)
    // Add 3
    .map(|x| x + 3)
    .collect::<Vec<i32>>();
```

Besides, it seems a bit weird that this piece of code generates a chaining hint:

```rust
let vec = (0..10)



    .collect::<Vec<i32>>();
```

But this one doesn't:

```rust
let vec = (0..10)

    // This is a comment

    .collect::<Vec<i32>>();
```

### CODE REVIEW

File: crates/ide/src/inlay_hints.rs [hunk 1]
```diff
@@ -983,6 +988,7 @@ struct C;
 fn main() {
     let c = A(B(C))
         .into_b() // This is a comment
+        // This is another comment
         .into_c();
 }
```
Comment (Veykril) [inline crates/ide/src/inlay_hints.rs:993]: this comment actually doesn't affect the test as the last line, the `.into_c();` doesn't trigger chaining hints given its the last call. The comment should instead be above the `.into_b()` call.

Comment (unratito) [reply crates/ide/src/inlay_hints.rs:993]: It affects the test in the following way:

- Without the added comment, two chaining hints are generated
- With the added comment and the current implementation, only one chaining hint is generated
- With the added comment and the modified implementation, two chaining hints are generated again (i.e., this is how I propose it should be)

Comment (Veykril) [reply crates/ide/src/inlay_hints.rs:993]: Oh I misunderstood the problem, I thought the comment affected the following lines. Yes then you are correct 👍 Was a bit confused as nothing changed in the test output(expected the TextRanges to change)

**Comment by Veykril** [comment]
Thanks!
bors r+

**Pull request merged**